src/Orleans/AssemblyLoader/AssemblyLoader.cs

```diff
@@ -319,9 +319,16 @@ private bool ReflectionOnlyLoadAssembly(string pathName, out Assembly assembly,
             {
                 if (SimulateReflectionOnlyLoadFailure)
                     throw NewTestUnexpectedException();
-                
+
                 assembly = Assembly.ReflectionOnlyLoadFrom(pathName);
             }
+            catch (BadImageFormatException)
+            {
+                // skip assemblies that are don't target the correct platform or are non-clr assemblies.
+                complaints = null;
```

Comment:
Interesting. You could possibly work out the compatibility by checking both the loaded assembly's `AssemblyName`'s `ProcessorArchitecture` against `Environment.Is64BitProcess`

![image](https://cloud.githubusercontent.com/assets/203839/21624010/2f882712-d259-11e6-8847-308b064701e1.png)